pages/sitemap/SitemapHandler.inc.php

```diff
@@ -70,50 +70,111 @@ function _createJournalSitemap() {
+		}
 		// Search
 		XMLCustomWriter::appendChild($root, $this->_createUrlTree($doc, Request::url($journal->getPath(), 'search')));
 		XMLCustomWriter::appendChild($root, $this->_createUrlTree($doc, Request::url($journal->getPath(), 'search', 'authors')));
-		XMLCustomWriter::appendChild($root, $this->_createUrlTree($doc, Request::url($journal->getPath(), 'search', 'titles')));
 		// Issues
-		XMLCustomWriter::appendChild($root, $this->_createUrlTree($doc, Request::url($journal->getPath(), 'issue', 'current')));
-		XMLCustomWriter::appendChild($root, $this->_createUrlTree($doc, Request::url($journal->getPath(), 'issue', 'archive')));
-		$publishedIssues = $issueDao->getPublishedIssues($journalId);
-		while ($issue = $publishedIssues->next()) {
-			XMLCustomWriter::appendChild($root, $this->_createUrlTree($doc, Request::url($journal->getPath(), 'issue', 'view', $issue->getId())));
-			// Articles for issue
-			$articles = $publishedArticleDao->getPublishedArticles($issue->getId());
-			foreach($articles as $article) {
-				// Abstract
-				XMLCustomWriter::appendChild($root, $this->_createUrlTree($doc, Request::url($journal->getPath(), 'article', 'view', array($article->getBestArticleId()))));
-				// Galley files
-				$galleys = $galleyDao->getBySubmissionId($article->getId());
-				while ($galley = $galleys->next()) {
-					XMLCustomWriter::appendChild($root, $this->_createUrlTree($doc, Request::url($journal->getPath(), 'article', 'view', array($article->getBestArticleId(), $galley->getId()))));
+		if ($journal->getSetting('publishingMode') != PUBLISHING_MODE_NONE) {
+			$issueCurrentUrl = Request::url($journal->getPath(), 'issue', 'current');
+			$issueArchiveUrl = Request::url($journal->getPath(), 'issue', 'archive');
+			if (!in_array($issueCurrentUrl, $menuURLs)) {
+				XMLCustomWriter::appendChild($root, $this->_createUrlTree($doc, $issueCurrentUrl));
+			}
+			if (!in_array($issueArchiveUrl, $menuURLs)) {
+				XMLCustomWriter::appendChild($root, $this->_createUrlTree($doc, $issueArchiveUrl));
+			}
+			$publishedIssues = $issueDao->getPublishedIssues($journalId);
+			while ($issue = $publishedIssues->next()) {
+				XMLCustomWriter::appendChild($root, $this->_createUrlTree($doc, Request::url($journal->getPath(), 'issue', 'view', $issue->getId())));
+				// Articles for issue
+				$articles = $publishedArticleDao->getPublishedArticles($issue->getId());
+				foreach($articles as $article) {
+					// Abstract
+					XMLCustomWriter::appendChild($root, $this->_createUrlTree($doc, Request::url($journal->getPath(), 'article', 'view', array($article->getBestArticleId()))));
+					// Galley files
+					$galleys = $galleyDao->getBySubmissionId($article->getId());
+					while ($galley = $galleys->next()) {
+						XMLCustomWriter::appendChild($root, $this->_createUrlTree($doc, Request::url($journal->getPath(), 'article', 'view', array($article->getBestArticleId(), $galley->getId()))));
```

Comment:
Should we exclude remote URL galleys here?

Reply:
I thought we can consider them too, because the link is also provided on the article page -- first the article/view/... is called and then redirected.

Reply:
:+1: